### PR TITLE
AFP-311 initial draft

### PR DESCRIPTION
**Pull request by qburst-aparna**
Screen Shot 2019-05-23 at 1 48 55 pm
Screen Shot 2019-05-23 at 1 54 25 pm

### CODE REVIEW

File: packages/website/src/bin/page-generator/index.js [hunk 1]
```diff
@@ -355,8 +365,16 @@ module.exports = async function generatePages(
     pagesPath,
     wrappersPath: componentsPath,
   };
+  console.log('ppppppppppp', packagesPaths);
```
Comment (pyupew) [inline packages/website/src/bin/page-generator/index.js]: remove console.log

File: packages/website/src/components/navigation/all-packages-nav-content.tsx [hunk 2]
```diff
@@ -10,63 +10,104 @@ import { gridSize as gridSizeFn } from '@atlaskit/theme';
...
+import TreeNavContent, { arrayToTreeItems } from './tree-nav-content';
 
 const gridSize = gridSizeFn();
 
+export type Data = { packageId: string; homePath: string; parentId?: string }[];
 export type Props = {
-  data: { packageId: string; homePath: string }[];
+  data: Data;
 };
+export type Page = {
```
Comment (pyupew) [inline packages/website/src/components/navigation/all-packages-nav-content.tsx]: where is this type used?

File: packages/website/src/components/navigation/all-packages-nav-content.tsx [hunk 3]
```diff
@@ -10,63 +10,104 @@ import { gridSize as gridSizeFn } from '@atlaskit/theme';
...
-            ))}
-          </Group>
-        </div>
-      )}
-    </MenuSection>
-  </>
-);
+const AllPackagesNavContent = ({ data }: Props) => {
+  const dataToConvert = data.filter(y => y.parentId);
+  const packagesWithParentIds: any = [];
```
Comment (pyupew) [inline packages/website/src/components/navigation/all-packages-nav-content.tsx]: should this array be of the above Page type? It looks like it contains the same format of object

File: packages/website/src/bin/default-config.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ module.exports = {
     name: 'Docs',
     description: 'View the documentation for this project',
   },
-  packages: ['./packages/*'],
+  packages: ['./packages/**/*'],
```
Comment (Noviny) [inline packages/website/src/bin/default-config.js]: Can we call out this change in a changeset?

WHAT the change is
WHY the change was made
HOW a consumer should update their code

In this case, the consumer (may) need to update their code.

File: .changeset/ecb02b75/changes.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+- Added the support for multi level packages and their nav items grouping.
+- This is necessary because without grouping the nested packages also falls in the root level flat structure in nav which makes package discoverability difficult.
+- For the multi level packages to show up in the website, consumers need to specify packages: ['./packages/**/*'] in the docs.config.js
```
Comment (Noviny) [inline .changeset/ecb02b75/changes.md]: Can we note that this is also now the default behavior?

File: .changeset/ecb02b75/changes.md [hunk 6]
```diff
@@ -1,3 +1,6 @@
 - Added the support for multi level packages and their nav items grouping.
 - This is necessary because without grouping the nested packages also falls in the root level flat structure in nav which makes package discoverability difficult.
 - For the multi level packages to show up in the website, consumers need to specify packages: ['./packages/**/*'] in the docs.config.js
+- This is being set as the default configuration of the website. If need to override consumers can specify packages: ['./packages/*']
+  or any other location according to their requirements.
```
Comment (Noviny) [inline .changeset/ecb02b75/changes.md]: Brilliant
